tasks/eva_3229/gather_counts_from_count_service.py

```diff
@@ -0,0 +1,151 @@
+                    elif v['clustered_variants_rs_split'] > 0 and v['clustered_variants_merge_operations'] == 0:
+                        results_by_assembly[assembly][metric] += count
+                    # If clustered variants are created and deprecated in a single step, probably a remediation that
+                    # included "resurrecting" valid RS - will count as creation
+                    elif v['clustered_variants_deprecated'] > 0:
+                        results_by_assembly[assembly][metric] += count
+                    # Otherwise this is probably creation of clustered variants due to remapping
+                    else:
+                        results_by_assembly[assembly]['clustered_variants_remapped'] += count
+                else:
+                    results_by_assembly[assembly][metric] += count
+        self.report(results_by_assembly)
+
+    def get_clustering_counts_between_dates_with_job_tracker(self, start_date, end_date):
+        """Get more granular per-assembly clustering counts between dates using the counts service and using the job
+        tracker to directly disambiguate different steps.
+        Dates are strings of the form YYYY-MM-DD (inclusive of start, exclusive of end)."""
+        created_rs_steps = ['CLUSTERING_NON_CLUSTERED_VARIANTS_FROM_MONGO_STEP', 'STUDY_CLUSTERING_STEP']
+        split_steps = ['PROCESS_RS_SPLIT_CANDIDATES_STEP']
+        remapped_rs_steps = ['CLUSTERING_CLUSTERED_VARIANTS_FROM_MONGO_STEP']
+        merge_steps = ['PROCESS_RS_MERGE_CANDIDATES_STEP']
+        deprecated_rs_steps = ['DEPRECATE_STUDY_SUBMITTED_VARIANTS_STEP', 'stepsForEVA3101Deprecation',
+                               'stepsForSSDeprecation']
+        all_steps_joined = ",".join(f"'{step}'" for step in created_rs_steps + remapped_rs_steps + deprecated_rs_steps
+                                    + split_steps + merge_steps)
+        with self.get_job_tracker_connection_handle() as jt_conn:
+            jt_query = (
+                "SELECT step_name, start_time, end_time, status FROM batch_step_execution "
+                f"WHERE start_time >= '{start_date}' AND end_time < '{end_date}' "
+                f"AND step_name IN ({all_steps_joined}) "
+            )
+            jt_results = get_all_results_for_query(jt_conn, jt_query)
+
+        results_by_assembly = defaultdict(Counter)
+        # Construct query that keeps track of step name and status for each time range
+        step_name_case = ""
+        status_case = ""
+        for step_name, start_time, end_time, status in jt_results:
+            step_name_case += f" WHEN timestamp >= '{start_time}' AND timestamp <= '{end_time}' THEN '{step_name}'"
+            status_case += f" WHEN timestamp >= '{start_time}' AND timestamp <= '{end_time}' THEN '{status}'"
+        query = (
+            f"SELECT identifier->>'assembly' as assembly, metric, count, "
+            f"CASE {step_name_case} END step_name, "
+            f"CASE {status_case} END status "
```

Comment:
I don't understand the purpose of the `status_case` and it does not seem to be used anywhere.

Reply:
You're right it's not being used, removed.